### ISSUE TITLE
Installing syslinux stuck at 0%

### ISSUE THREAD

**Issue by alindt** [mbusb/multibootusb#111]
**Using MultiBootUSB 8.5.0**

```bash
$  apt show python3-multibootusb 
Package: python3-multibootusb
Version: 8.5.0-1
Status: install ok installed
```

I've tried several ISOs and they fail at "Installing syslinux".

**ISO 1 (DamnSmallLinux)**
**(LE: fixed in 	578c61b)**

```
$  sudo multibootusb
Starting multibootusb GUI...
Using pyudev for detecting USB drives...
Cleaning old multibootusb directory...
Extracting syslinux to multibootusb directory...
Host OS is 64 bit...
Extracting syslinux 64 bit...
Extracting syslinux modules to multibootusb directory...
Persistence is not available for dslinux.iso
USB Disk is /dev/sdb1
USB Label is rescue
USB UUID is 8470-2746
USB Mount path is /media/alin/rescue
Total size of the disk is 7.5G
Total used size  is 305.5M
Total size left on the disk is 7.2G
FileSystem is vfat
Vendor is TDKMedia
Model is TF090_Drive
Name of the ISO file is dslinux.iso
Distro type detected is knoppix
multibootus directory already exist. Not copying.
Persistence choosen is 0 MB
Installing dslinux.iso on /media/alin/rescue/multibootusb/dslinux             |
ISO extracted successfully. Sync is in progress...########################### |
Distro extraction completed...
/media/alin/rescue/multibootusb/dslinux/boot/isolinux/knoppix.bs
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/scripts/mbusb_gui.py", line 210, in install_syslinux
    syslinux_distro_dir(config.usb_disk, config.iso_link, config.distro)
  File "/usr/lib/python3/dist-packages/scripts/syslinux.py", line 139, in syslinux_distro_dir
    syslinux_path = os.path.join(multibootusb_host_dir(), "syslinux", "bin", "syslinux") + config.syslinux_version
TypeError: Can't convert 'NoneType' object to str implicitly
Closing multibootusb...
```

**ISO 2 (kaspersky_rescue_10)**

```
$  sudo multibootusb
Starting multibootusb GUI...
Using pyudev for detecting USB drives...
Cleaning old multibootusb directory...
Extracting syslinux to multibootusb directory...
Host OS is 64 bit...
Extracting syslinux 64 bit...
Extracting syslinux modules to multibootusb directory...
Using pyudev for detecting USB drives...
 This disc don't support RRIP
Persistence is not available for kav_rescue_10.iso
USB Disk is /dev/sdb1
USB Label is rescue
USB UUID is 8470-2746
USB Mount path is /media/alin/rescue
Total size of the disk is 7.5G
Total used size  is 4.0K
Total size left on the disk is 7.5G
FileSystem is vfat
Vendor is TDKMedia
Model is TF090_Drive
Name of the ISO file is kav_rescue_10.iso
Distro type detected is kaspersky
Copying multibootusb directory to /media/alin/rescue
Persistence choosen is 0 MB
Copying multibootusb directory to /media/alin/rescue                          |
Installing kav_rescue_10.iso on /media/alin/rescue/multibootusb/kav_rescue_10
ISO extracted successfully. Sync is in progress...############################|
Distro extraction completed...
Distro does not use isolinux for booting ISO.

Executing ==> /home/alin/.multibootusb/syslinux/bin/syslinux4 -i -d multibootusb /dev/sdb1


Default syslinux install is success...

1+0 records in
1+0 records out
440 bytes copied, 0.000793594 s, 554 kB/s

mbr install is success...


Checking boot flag on /dev/sdb

Disk /dev/sdb already has boot flag.

Updating distro specific config files...
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/scripts/mbusb_gui.py", line 212, in install_syslinux
    update_distro_cfg_files(config.iso_link, config.usb_disk, config.distro, config.persistence)
  File "/usr/lib/python3/dist-packages/scripts/update_cfg_file.py", line 88, in update_distro_cfg_files
    os.path.join(usb_mount, 'multibootusb', iso_basename(iso_link), 'kaspersky.cfg'))
  File "/usr/lib/python3.5/shutil.py", line 114, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/multibootusb/data/multibootusb/syslinux.cfg'
Closing multibootusb...
```
```
$  sudo ls -lR /usr/share/multibootusb
/usr/share/multibootusb:
total 4
drwxr-xr-x 3 root root 4096 Apr  2 20:52 data

/usr/share/multibootusb/data:
total 8
drwxr-xr-x 6 root root 4096 Apr  2 20:52 tools
-rw-r--r-- 1 root root    7 Feb  5 12:02 version.txt

/usr/share/multibootusb/data/tools:
total 24
drwxr-xr-x 2 root root 4096 Apr  2 20:52 dd
-rw-r--r-- 1 root root  440 Feb  5 12:02 mbr.bin
drwxr-xr-x 2 root root 4096 Apr  2 20:52 mkfs
drwxr-xr-x 2 root root 4096 Apr  2 20:52 multibootusb
-rw-r--r-- 1 root root 3047 Feb  5 12:02 multibootusb.png
drwxr-xr-x 2 root root 4096 Apr  2 20:52 syslinux

/usr/share/multibootusb/data/tools/dd:
total 372
-rw-r--r-- 1 root root 355328 Feb  5 12:02 dd.exe
-rw-r--r-- 1 root root  21190 Feb  5 12:02 diskio.dll

/usr/share/multibootusb/data/tools/mkfs:
total 112
-rw-r--r-- 1 root root 114688 Feb  5 12:02 mke2fs.exe

/usr/share/multibootusb/data/tools/multibootusb:
total 516
-rw-r--r-- 1 root root   2052 Feb  5 12:02 bg.png
-rw-r--r-- 1 root root  20576 Feb  5 12:02 chain.c32
-rw-r--r-- 1 root root      0 Feb  5 12:02 extlinux.cfg
-rw-r--r-- 1 root root 236977 Feb  5 12:02 grub.exe
-rw-r--r-- 1 root root  25244 Feb  5 12:02 memdisk
-rw-r--r-- 1 root root  60928 Feb  5 12:02 menu.c32
-rw-r--r-- 1 root root      2 Feb  5 12:02 menu.lst
-rw-r--r-- 1 root root    990 Feb  5 12:02 syslinux.cfg
-rw-r--r-- 1 root root 162860 Feb  5 12:02 vesamenu.c32

/usr/share/multibootusb/data/tools/syslinux:
total 3872
-rw-r--r-- 1 root root  610253 Feb  5 12:02 syslinux_linux_64.zip
-rw-r--r-- 1 root root  648542 Feb  5 12:02 syslinux_linux.zip
-rw-r--r-- 1 root root 2336932 Feb  5 12:02 syslinux_modules.zip
-rw-r--r-- 1 root root  361728 Feb  5 12:02 syslinux_windows.zip
```

**Comment by jphein** [comment]
This happens to me fairly often. I think it points to a failing USB drive.

**Comment by alindt** [comment]
No, this one is a different problem:
```
FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/multibootusb/data/multibootusb/syslinux.cfg'
```

**Comment by mbusb** [comment]
@alindt I have fixed the first issue (DSL) and commited in to devel branch. 

Regarding your 2nd issue of `FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/multibootusb/data/multibootusb/syslinux.cfg'
Closing multibootusb...` and your log report, I can see something wrong with the installation. 'syslinux.cfg' shipped with the package resides in `/multibootusb/data/multibootusb` directory and not in `/usr/share/multibootusb/data/tools/multibootusb`. Probably issue with the setup.py script. Will check it out.
